python/ray/train/xgboost/xgboost_trainer.py

```diff
@@ -79,34 +72,36 @@ class XGBoostTrainer(GBDTTrainer):
         **train_kwargs: Additional kwargs passed to ``xgboost.train()`` function.
     """
 
-    _dmatrix_cls: type = xgboost_ray.RayDMatrix
-    _ray_params_cls: type = xgboost_ray.RayParams
-    _tune_callback_checkpoint_cls: type = TuneReportCheckpointCallback
     _default_ray_params: Dict[str, Any] = {
         "num_actors": 1,
         "cpus_per_actor": 1,
         "gpus_per_actor": 0,
     }
     _init_model_arg_name: str = "xgb_model"
 
-    @staticmethod
-    def get_model(checkpoint: Checkpoint) -> xgboost.Booster:
+    def __init__(self, *args, **kwargs):
+        # TODO(justinvyu): Fix circular import by moving xgboost_ray into ray
+        import xgboost_ray
+
+        self._dmatrix_cls: type = xgboost_ray.RayDMatrix
+        self._ray_params_cls: type = xgboost_ray.RayParams
+        self._tune_callback_checkpoint_cls: type = (
+            xgboost_ray.tune.TuneReportCheckpointCallback
+        )
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def get_model(
+        cls,
+        checkpoint: Checkpoint,
+    ) -> xgboost.Booster:
```

Comment:
Update the docstring to suggest users to use `RayTrainReportCallback` directly if they need any customization (e.g. if they used a different path)?

Reply:
`XGBoostTrainer` currently creates the callback for you and doesn't customize the `filename`, so I think we don't need to cover the custom case here.